Change to integers for all table primary keys

usask-rc opened the pull request:
- gmeta batch size lowered due to Globus server side constraints
- renamed column 'date' to 'pub_date' in records table
- switched to size-based log rolling since date-based was not working
- repositories that differ only in set now correctly identified as separate

Commit:
Change to integers for all table primary keys

axfelix commented on harvester/DBInterface.py:
```diff
@@ -26,34 +26,26 @@ def __init__(self, params):
+			cur.execute("create index if not exists geospatial_by_record on geospatial(record_id)")
+			cur.execute("create index if not exists domain_metadata_by_record on domain_metadata(record_id)")
+			cur.execute("create unique index if not exists records_by_repository on records (repository_id, local_identifier)")
 
```
This is a lot harder to read than before, but not a huge problem; I don't think the original code was anywhere close to PEP8 so I don't have much of a defense.

axfelix commented on harvester/DBInterface.py:
```diff
@@ -77,107 +69,199 @@ def _prep(self, statement):
+				# Create new repo record
+				try:
+					self.logger.debug("This repo did not exist in the database; adding")
+					if self.dbtype == "postgres":
```
Is it necessary to split this logic by postgres/SQLite? Lite should support the fetchone() method...

axfelix replied:
Also, to clarify: have you tested this on psql and SQLite already yourself?

usask-rc replied:
SQLite doesn't support the RETURNING keyword in the SQL, and the Postgres python library doesn't support lastrowid, from what I've read.

axfelix commented on harvester/DBInterface.py:
```diff
@@ -77,107 +69,199 @@ def _prep(self, statement):
+			source_url = ""
+			if 'dc:source' in record:
+				source_url = record["dc:source"]
+			if record["record_id"] == 0:
```
I'm a little ambivalent about returning 0's for repo/repository ID lookups -- None's should be faster and more logical to test for?

usask-rc replied:
Agreed, will change to None.

axfelix commented on harvester/DBInterface.py:
```diff
@@ -77,107 +69,199 @@ def _prep(self, statement):
 				for creator in record["creator"]:
 					try:
-						cur.execute(self._prep("INSERT INTO creators (local_identifier, repository_url, creator, is_contributor) VALUES (?,?,?,?)"), (record["identifier"], repository_url, creator, 0))
+						cur.execute(self._prep("DELETE from creators where record_id = ? and is_contributor=0"), (record["record_id"],))
```
Is this approach necessary? Deleting and removing the corresponding rows on every run?

axfelix commented:
Hi Todd,

A few comments above but nothing major, generally looks great. I'm having trouble testing due to DNS issues here this morning, but I imagine those will be resolved soon.

axfelix commented:
OK, managing to trigger some issues around the < 1 checking, that's another point in favour of using None's:

    Traceback (most recent call last):
      File "globus_harvester.py", line 90, in <module>
        exporter.export_to_file(configs['export_format'], configs['export_filepath'], configs['export_batch_size'], False, configs['temp_filepath'])
      File "/home/axfelix/Desktop/globus_oai/harvester/Exporter.py", line 260, in export_to_file
        gmeta_block = self._generate_gmeta(export_batch_size, export_filepath, temp_filepath, only_new_records)
      File "/home/axfelix/Desktop/globus_oai/harvester/Exporter.py", line 95, in _generate_gmeta
        if (record['title'] < 1):
    TypeError: '<' not supported between instances of 'str' and 'int'

usask-rc commented:
The error on line 95 was me missing the len() call. It was in there at one time but I was messing with other logic, took it out, forgot to put back in.  The code as it stands doesn't make any sense at all, I'll fix it.

Commit:
Fix a few errors

Commit:
Fix a few errors

axfelix commented:
Is it me, or are the logo image URLs being written into every URL field (replacing the actual item landing pages) in the output JSON with this branch currently?

usask-rc commented:
There is something wrong URLs in a few places, yes.  I hadn't reviewed the output data yet, I'll look now.

Commit:
Trap regex error in CKAN
Fix column ordering in OAI
Do not munge YYYY dates
Remove more unneeded output columns

axfelix commented:
Almost there ... I'm getting geospatial polygons output in the Gmeta with only a single point preserved.

usask-rc commented:
What item?  I looked at a sample item 0019addf-701f-5bae-be7d-7566bbb5a7cf and it looked OK to me:

```
                    "nrdr:geospatial": [
                        {
                            "geometry": {
                                "coordinates": [
                                    [
                                        -129.94952,
                                        68.34952
                                    ],
                                    [
                                        -129.94952,
                                        69.32243
                                    ],
                                    [
                                        -126.98836,
                                        68.34952
                                    ],
                                    [
                                        -126.98836,
                                        69.32243
                                    ]
                                ],
                                "type": "Polygon"
                            },
                            "type": "Feature"
                        }
                    ],
```

Commit:
Remove excess date debugging; use correct var for OAI error

axfelix commented:
https://www.polardata.ca/pdcsearch/PDCSearch.jsp?doi_id=80 is one, but they're actually all coming out this way for me on this branch...

usask-rc commented:
Ah, it's the SQL to update the db, losing some of the coords in translation.  I'll check why.

usask-rc commented:
Nope, red herring.  Must be the "delete all records before inserting new ones", but then not re-inserting all the new ones back again.  I'll check that!

Commit:
Fix some table column names

Commit:
Check in the update sql

Commit:
Fix position of record deletion; Stop error logging from generating new error

axfelix commented:
OK, just getting back to this now before the long weekend ... we probably don't need the update SQL in the project root moving forward, but as long as this is all working now I'd be happy to merge and run it for the time being.

axfelix commented:
Can't find any more issues -- merging this!
